File: app/src/main/java/com/ayvytr/gank/MainActivity.java
package com.ayvytr.gank;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.constraint.Group;
import android.support.design.widget.BottomNavigationView;
import android.support.design.widget.NavigationView;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.view.GravityCompat;
import android.support.v4.view.ViewPager;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.FrameLayout;

import com.alibaba.android.arouter.launcher.ARouter;
import com.ayvytr.commonlibrary.GankType;
import com.ayvytr.commonlibrary.constant.KnowledgeConstant;
import com.ayvytr.easykotlin.ui.ViewKt;
import com.ayvytr.girl.view.fragment.GirlsFragment;
import com.ayvytr.knowledge.view.fragment.AndroidFragment;
import com.ayvytr.knowledge.view.fragment.ClassifyGankFragment;
import com.ayvytr.knowledge.view.fragment.GankHistoryFragment;
import com.ayvytr.mvpbase.IPresenter;
import com.ayvytr.mvprxlifecycle.BaseMvpActivity;
import com.ayvytr.settings.SettingsFragment;

import butterknife.BindView;
import butterknife.ButterKnife;

public class MainActivity extends BaseMvpActivity
        implements NavigationView.OnNavigationItemSelectedListener {

    @BindView(R.id.toolbar)
    Toolbar mToolbar;
    @BindView(R.id.nav_view)
    NavigationView mNavView;
    @BindView(R.id.drawer_layout)
    DrawerLayout mDrawerLayout;
    @BindView(R.id.bottom_navigation)
    BottomNavigationView mBottomNavigation;
    @BindView(R.id.vp)
    ViewPager mVp;
    @BindView(R.id.flContainer)
    FrameLayout mFlContainer;
    @BindView(R.id.group)
    Group mGroup;

    private Fragment[] mFragments = {
            AndroidFragment.newInstance(GankType.ANDROID),
            new GirlsFragment(),
            new SettingsFragment()
    };

    @Override
    protected IPresenter getPresenter() {
        return null;
    }

    @Override
    public void onBackPressed() {
        if(mDrawerLayout.isDrawerOpen(GravityCompat.START)) {
            mDrawerLayout.closeDrawer(GravityCompat.START);
        } else {
            super.onBackPressed();
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if(id == R.id.menu_search) {
            ARouter.getInstance().build(KnowledgeConstant.SEARCH).navigation(getContext());
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @SuppressWarnings("StatementWithEmptyBody")
    @Override
    public boolean onNavigationItemSelected(MenuItem item) {
        // Handle navigation view item clicks here.
        int id = item.getItemId();

        ViewKt.show(mFlContainer, id != R.id.nav_home);
        ViewKt.show(mGroup, id == R.id.nav_home);

        if(id == R.id.nav_home) {
            ViewKt.hide(mFlContainer);
        } else if(id == R.id.nav_gank_history) {
            FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
            ft.replace(R.id.flContainer, new GankHistoryFragment());
            ft.commit();
        } else if(id == R.id.nav_gank_of_type) {
            FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
            ft.replace(R.id.flContainer, new ClassifyGankFragment());
            ft.commit();
        } else if(id == R.id.nav_manage) {

        } else if(id == R.id.nav_share) {

        } else if(id == R.id.nav_send) {

        }

        mDrawerLayout.closeDrawer(GravityCompat.START);
        return true;
    }

    @Override
    public void initExtra() {

    }

    @Override
    public void initView(@Nullable Bundle savedInstanceState) {
        ButterKnife.bind(this);
        setSupportActionBar(mToolbar);
        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
                this, mDrawerLayout, mToolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
        mDrawerLayout.addDrawerListener(toggle);
        toggle.syncState();
        mNavView.setNavigationItemSelectedListener(this);

        mVp.setAdapter(new FragmentPagerAdapter(getSupportFragmentManager()) {
            @Override
            public Fragment getItem(int position) {
                return mFragments[position];
            }

            @Override
            public int getCount() {
                return mFragments.length;
            }
        });

        mBottomNavigation.setOnNavigationItemSelectedListener(
                new BottomNavigationView.OnNavigationItemSelectedListener() {
                    @Override
                    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                        switch(item.getItemId()) {
                            case R.id.nv_android:
                                mVp.setCurrentItem(0);
                                return true;
                            case R.id.nv_girls:
                                mVp.setCurrentItem(1);
                                return true;
                            case R.id.nv_settings:
                                mVp.setCurrentItem(2);
                                return true;
                            default:
                                return false;
                        }
                    }
                });
        mVp.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
            }

            @Override
            public void onPageSelected(int position) {
                switch(position) {
                    case 0:
                        mBottomNavigation.setSelectedItemId(R.id.nv_android);
                        break;
                    case 1:
                        mBottomNavigation.setSelectedItemId(R.id.nv_girls);
                        break;
                    default:
                        mBottomNavigation.setSelectedItemId(R.id.nv_settings);
                        break;
                }
            }

            @Override
            public void onPageScrollStateChanged(int state) {
            }
        });
    }

    @Override
    public void initData(@Nullable Bundle savedInstanceState) {

    }

    @Override
    public int getContentViewRes() {
        return R.layout.activity_main;
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mBottomNavigation.setOnNavigationItemSelectedListener(null);
        mVp.setAdapter(null);
        mFragments = null;
    }
}

File: app/src/main/java/com/ayvytr/gank/MainActivity.kt
package com.ayvytr.gank

import android.os.Bundle
import android.support.design.widget.NavigationView
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentPagerAdapter
import android.support.v4.view.GravityCompat
import android.support.v4.view.ViewPager
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v7.widget.Toolbar
import android.view.Menu
import android.view.MenuItem
import com.alibaba.android.arouter.launcher.ARouter
import com.ayvytr.commonlibrary.GankType
import com.ayvytr.commonlibrary.constant.KnowledgeConstant
import com.ayvytr.easykotlin.ui.hide
import com.ayvytr.easykotlin.ui.show
import com.ayvytr.girl.view.fragment.GirlsFragment
import com.ayvytr.knowledge.view.fragment.AndroidFragment
import com.ayvytr.knowledge.view.fragment.ClassifyGankFragment
import com.ayvytr.knowledge.view.fragment.GankHistoryFragment
import com.ayvytr.mvp.IPresenter
import com.ayvytr.rxlifecycle.BaseMvpActivity
import com.ayvytr.settings.SettingsFragment
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.content_main.*

class MainActivity : BaseMvpActivity<IPresenter>(), NavigationView.OnNavigationItemSelectedListener {

    private var mFragments: Array<Fragment>? = arrayOf(AndroidFragment.newInstance(GankType.ANDROID), GirlsFragment(), SettingsFragment())

    private lateinit var toolbar:Toolbar

    override fun getPresenter(): IPresenter? {
        return null
    }

    override fun onBackPressed() {
        if (drawer_layout!!.isDrawerOpen(GravityCompat.START)) {
            drawer_layout!!.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        val id = item.itemId


        if (id == R.id.menu_search) {
            ARouter.getInstance().build(KnowledgeConstant.SEARCH).navigation(context)
            return true
        }

        return super.onOptionsItemSelected(item)
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        // Handle navigation view item clicks here.
        val id = item.itemId

        flContainer.show(id != R.id.nav_home)
        group.show(id == R.id.nav_home)

        if (id == R.id.nav_home) {
            flContainer.hide()
        } else if (id == R.id.nav_gank_history) {
            val ft = supportFragmentManager.beginTransaction()
            ft.replace(R.id.flContainer, GankHistoryFragment())
            ft.commit()
        } else if (id == R.id.nav_gank_of_type) {
            val ft = supportFragmentManager.beginTransaction()
            ft.replace(R.id.flContainer, ClassifyGankFragment())
            ft.commit()
        } else if (id == R.id.nav_more) {
        }

        drawer_layout!!.closeDrawer(GravityCompat.START)
        return true
    }

    override fun initExtra() {

    }

    override fun initView(savedInstanceState: Bundle?) {
        toolbar = findViewById(R.id.toolbar)
        setSupportActionBar(toolbar)
        val toggle = ActionBarDrawerToggle(
            this, drawer_layout, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close)
        drawer_layout!!.addDrawerListener(toggle)
        toggle.syncState()
        nav_view!!.setNavigationItemSelectedListener(this)

        vp!!.adapter = object : FragmentPagerAdapter(supportFragmentManager) {
            override fun getItem(position: Int): Fragment {
                return mFragments!![position]
            }

            override fun getCount(): Int {
                return mFragments!!.size
            }
        }

        bottom_navigation!!.setOnNavigationItemSelectedListener { item ->
            when (item.itemId) {
                R.id.nv_android  -> {
                    vp!!.currentItem = 0
                    true
                }
                R.id.nv_girls    -> {
                    vp!!.currentItem = 1
                    true
                }
                R.id.nv_settings -> {
                    vp!!.currentItem = 2
                    true
                }
                else             -> false
            }
        }
        vp!!.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {}

            override fun onPageSelected(position: Int) {
                when (position) {
                    0    -> bottom_navigation!!.selectedItemId = R.id.nv_android
                    1    -> bottom_navigation!!.selectedItemId = R.id.nv_girls
                    else -> bottom_navigation!!.selectedItemId = R.id.nv_settings
                }
            }

            override fun onPageScrollStateChanged(state: Int) {}
        })
    }

    override fun initData(savedInstanceState: Bundle?) {

    }

    override fun getContentViewRes(): Int {
        return R.layout.activity_main
    }

    override fun onDestroy() {
        super.onDestroy()
        bottom_navigation!!.setOnNavigationItemSelectedListener(null)
        vp!!.adapter = null
        mFragments = null
    }
}
